pkgs/applications/gis/qgis/unwrapped.nix

```diff
@@ -132,31 +138,76 @@ in mkDerivation rec {
     "-DWITH_3D=True"
-    "-DWITH_PDAL=TRUE"
   ] ++ lib.optional (!withWebKit) "-DWITH_QTWEBKIT=OFF"
+    ++ lib.optional stdenv.isLinux "-DWITH_PDAL=TRUE"
     ++ lib.optional withGrass (let
         gmajor = lib.versions.major grass.version;
         gminor = lib.versions.minor grass.version;
       in "-DGRASS_PREFIX${gmajor}=${grass}/grass${gmajor}${gminor}"
     );
 
   dontWrapGApps = true; # wrapper params passed below
+  dontWrapQtApps = stdenv.isDarwin;
 
-  postFixup = lib.optionalString withGrass ''
+  postFixup = lib.optionalString (withGrass && stdenv.isLinux) ''
     # grass has to be availble on the command line even though we baked in
     # the path at build time using GRASS_PREFIX.
     # using wrapGAppsHook also prevents file dialogs from crashing the program
     # on non-NixOS
     wrapProgram $out/bin/qgis \
       "''${gappsWrapperArgs[@]}" \
       --prefix PATH : ${lib.makeBinPath [ grass ]}
+  '' + lib.optionalString stdenv.isDarwin ''
+    mkdir -p $out/Applications
+    mkdir -p $out/bin
+    mv $out/QGIS.app $out/Applications/
+    ln -s $out/Applications/QGIS.app/Contents/MacOS/QGIS $out/bin/qgis
+    export SHORT_VERSION=$(echo "${version}" | awk 'BEGIN{FS=OFS="."} NF--')
+    echo "Short version: $SHORT_VERSION"
+    for f in $out/Applications/QGIS.app/Contents/MacOS/lib/libqgis_app.${version}.dylib \
+             $out/Applications/QGIS.app/Contents/MacOS/lib/libqgispython.${version}.dylib \
+             $out/Applications/QGIS.app/Contents/MacOS/QGIS \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_3d.framework/Versions/$SHORT_VERSION/qgis_3d \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_analysis.framework/Versions/$SHORT_VERSION/qgis_analysis \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_core.framework/Versions/$SHORT_VERSION/qgis_core \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_gui.framework/Versions/$SHORT_VERSION/qgis_gui \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_native.framework/Versions/$SHORT_VERSION/qgis_native \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgisgrass8.framework/Versions/$SHORT_VERSION/qgisgrass8 \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_3d.framework/qgis_3d \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_analysis.framework/qgis_analysis \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_core.framework/qgis_core \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_gui.framework/qgis_gui \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgis_native.framework/qgis_native \
+             $out/Applications/QGIS.app/Contents/Frameworks/qgisgrass8.framework/qgisgrass8
+      do
+        echo "Manually running install_name_tool on $f"
+        install_name_tool -change "@loader_path/../lib/libqscintilla2_qt5.dylib" "${qscintilla}/lib/libqscintilla2_qt5.dylib" $f
```

Comment:
how did you get this list ? can it be parsed and looped over ? it should be more resistant to future upgrade

Reply:
This was, unfortunately, done by me repeatedly building and running the .app, having it crash due to a linking error, and repeating over and over. I did at least simplify it so it can loop over the dylibs I know about. It seems like the [original build script does it with a little less targeting](https://github.com/qgis/QGIS-Mac-Packager/blob/master/qgis_bundle/qgis_bundle.bash), but it's true that this is moderately fragile. I'm open to suggestions for how to improve it -- fixDarwinDylib names was too blunt an instrument and tried to operate on lots of files in $out, and wasn't able to deal with the Frameworks, so I ended up just doing this manually.